### ISSUE TITLE
Return null for missing property/leaf

### ISSUE THREAD

**Issue by brno32** [h2non/jsonpath-ng#82]
given

```python
[
    {"name": "Test Account"},
    {}
]
```

`$.[*].name` returns

```python
[
   "Test Account"
]
```

but I'd like for it to return 

```python
[
   "Test Account",
    None
]
```

https://jsonpath.herokuapp.com/ let's you decide whether or not to return `null` in this case. Does such a flag exist here?

**Comment by sadilet** [comment]
any updates on this? @brno32  eventually how did you solve it? I'm facing this issue now, thinking to override `find_or_create` method

**Comment by brno32** [comment]
I gave up on this library unfortunately. Overriding `find_or_create` might be a good approach

**Comment by Johnserf-Seed** [comment]
Yes, I had a similar problem. I need a field that doesn't necessarily exist, and when I use ```$.list[*].key[0]``` to match it only returns the found field instead of replacing its index with ```None``` !

Is there a good solution?
